server/api/crud/pipelines.py

```diff
@@ -433,3 +440,25 @@ def _get_experiment_id_from_run(run: dict) -> str:
             ):
                 return data.get("id", "")
         return ""
+
+    @staticmethod
+    def _filter_runs_by_name(runs: list, project_name: str, target_name: str) -> list:
+        """Filter runs by their name while ignoring the project string on them
+
+        :param runs: list of runs to be filtered
+        :param project_name: string to be disregarded while filtering
+        :param name: string that should be part of a valid run name
+        :return: filtered list of runs
+        """
+        if not target_name:
+            return runs
+
+        def filter_by(run):
+            run_name = run.get("name", "")
+            if project_name != "*":
+                run_name = run_name.replace(project_name + "-", "")
```

Comment:
At this point, each run has a project in it - we fill that as part fo the `list_pipelines` flow (in `_format_run`). So, instead of passing a project to this function, you can simply use each run's project name to perform the removal - this will work also for the `*` case.

Reply:
I didn't realize that. Thanks for pointing it out, Saar. I improved the code accordingly.